{
  try {
    run_inner();
  }
 catch (  Throwable t) {
    MonitorTest.errorException=t;
    MonitorTest.testThread.interrupt();
  }
}
